api/v1beta2/provider_types.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2022 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	"time"
+
+	"github.com/fluxcd/pkg/apis/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ProviderKind           string = "Provider"
+	GenericProvider        string = "generic"
+	GenericHMACProvider    string = "generic-hmac"
+	SlackProvider          string = "slack"
+	GrafanaProvider        string = "grafana"
+	DiscordProvider        string = "discord"
+	MSTeamsProvider        string = "msteams"
+	RocketProvider         string = "rocket"
+	GitHubDispatchProvider string = "githubdispatch"
+	GitHubProvider         string = "github"
+	GitLabProvider         string = "gitlab"
+	BitbucketProvider      string = "bitbucket"
+	AzureDevOpsProvider    string = "azuredevops"
+	GoogleChatProvider     string = "googlechat"
+	WebexProvider          string = "webex"
+	SentryProvider         string = "sentry"
+	AzureEventHubProvider  string = "azureeventhub"
+	TelegramProvider       string = "telegram"
+	LarkProvider           string = "lark"
+	Matrix                 string = "matrix"
+	OpsgenieProvider       string = "opsgenie"
+	AlertManagerProvider   string = "alertmanager"
+)
+
+// ProviderSpec defines the desired state of the Provider.
+type ProviderSpec struct {
+	// Type specifies which Provider implementation to use.
+	// +kubebuilder:validation:Enum=slack;discord;msteams;rocket;generic;generic-hmac;github;gitlab;bitbucket;azuredevops;googlechat;webex;sentry;azureeventhub;telegram;lark;matrix;opsgenie;alertmanager;grafana;githubdispatch;
+	// +required
+	Type string `json:"type"`
+
+	// Interval at which to reconcile the Provider with its Secret references.
+	// +kubebuilder:default="10m"
```

Comment:
This default value in a required field doesn't help. Not setting it still results in validation failure at creation time.
I see that we don't set a default for interval in SC and KC.

Reply:
Right, this should've been optional to preserver backwards compatibility. Done now in the latest commit. Thanks!